### PR TITLE
Boss bar support

### PR DESCRIPTION
**Pull request by Jens-Co**

### CODE REVIEW

**Comment by Jens-Co** [comment]
Loggers and reload needs to be implemented + papi (not really needed)

File: src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarHandler.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public static HashMap<Integer, String> color = new HashMap<>();
+    public static HashMap<Integer, String> text = new HashMap<>();
+    public static HashMap<Integer, String> style = new HashMap<>();
+
+    public void bossBarSetup(){
+        FileConfiguration config = GeyserHubMain.getInstance().getConfigManager().getFileConfiguration(ConfigId.MAIN);
+
+        int i = 1;
+        while(config.getString("BossBar.Bar" + i + ".Text") != null) {
```
Comment (Konicai) [inline src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarHandler.java]: Can we this this as the expression for the while loop:

`config.contains("BossBar.Bar" + i + ".Text") && config.contains("BossBar.Bar" + i + ".Color") && `

Comment (Konicai) [reply src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarHandler.java]: Also, should we be enforcing the keys to all be of the format ("Bar"+int) ? Does it really matter if they aren't?

File: src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarListener.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    public void onJoin(PlayerJoinEvent e){
+        for(Integer index : BossBarHandler.color.keySet()) {
+            if(getColor(BossBarHandler.color.get(index)) == null) {
+                BossBar bossBar = Bukkit.createBossBar("§cThis color is not available!", BarColor.WHITE, BarStyle.SOLID);
+                bossBar.addPlayer(e.getPlayer());
+            }else {
+                if(getBarStyle(BossBarHandler.style.get(index)) == null) {
+                    BossBar bossBar = Bukkit.createBossBar("§cThis BarType is not available!", BarColor.WHITE, BarStyle.SOLID);
+                    bossBar.addPlayer(e.getPlayer());
+                    return;
```
Comment (Konicai) [inline src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarListener.java:23]: probably should make a warn log in the console if the config isn't right, and check validity on setup of BosBarHandler

**Review by Konicai — COMMENTED** [review]
nice

File: src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarHandler.java (hunk 1 above)
Comment (Jens-Co) [reply src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarHandler.java]: Think we can yea, I'm still not sure if we should keep the loop who actually want 2 bossbars :p. But incase you want to keep it, for enforcement don't think is would matter that much, You can change it if you want to

File: src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarListener.java (hunk 2 above)
Comment (Jens-Co) [reply src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarListener.java:23]: Yea sounds good, for most parts like error on color or type we send an error that the format isn't good

File: src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarListener.java [hunk 3]
```diff
@@ -3,24 +3,23 @@
...
+    @EventHandler
     public void onJoin(PlayerJoinEvent e){
         for(Integer index : BossBarHandler.color.keySet()) {
             if(getColor(BossBarHandler.color.get(index)) == null) {
                 BossBar bossBar = Bukkit.createBossBar("§cThis color is not available!", BarColor.WHITE, BarStyle.SOLID);
                 bossBar.addPlayer(e.getPlayer());
             }else {
-
                 if(getBarStyle(BossBarHandler.style.get(index)) == null) {
                     BossBar bossBar = Bukkit.createBossBar("§cThis BarType is not available!", BarColor.WHITE, BarStyle.SOLID);
```
Comment (simonsoundara) [inline src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarListener.java]: you shouldn't use '§'

Comment (Konicai) [reply src/main/java/dev/projectg/geyserhub/module/bossBar/BossBarListener.java]: Use ChatColor if available

**Comment by Jens-Co** [comment]
Closed, 1.2.0 will get merged soon. ill make BossBar an revision branch to add all the left to do's from 1.2.0

**Comment by Jens-Co** [comment]
> Closed, 1.2.0 will get merged soon. ill make BossBar an revision branch to add all the left to do's from 1.2.0

ill update from main, planning to update spigot api to 1.17

**Pull request closed without merging**